calendar-webadmin/src/main/java/com/linagora/calendar/webadmin/ResourceRoutes.java

```diff
@@ -156,84 +182,149 @@ private ResourceDTO getResource(Request req) {
         return resourceDAO.findById(id)
             .flatMap(this::toDto)
             .blockOptional()
-            .orElseGet(() -> {
-                throw ErrorResponder.builder()
-                    .statusCode(HttpStatus.NOT_FOUND_404)
-                    .type(ErrorResponder.ErrorType.NOT_FOUND)
-                    .message("Resource do not exist")
-                    .cause(new RuntimeException())
-                    .haltError();
-            });
+            .orElseThrow(() -> resourceNotFound(id));
     }
 
     private String deleteResource(Request req, Response res) {
         ResourceId id = new ResourceId(req.params("id"));
 
-        try {
-            resourceDAO.softDelete(id).block();
-            res.status(204);
-            return "";
-        } catch (ResourceNotFoundException e) {
-            throw ErrorResponder.builder()
-                .statusCode(HttpStatus.NOT_FOUND_404)
-                .type(ErrorResponder.ErrorType.NOT_FOUND)
-                .message("Resource do not exist")
-                .cause(new RuntimeException())
-                .haltError();
-        }
+        return findNotDeletedResource(id)
+            .flatMap(currentResource -> resolveAdminUsernames(currentResource)
+                .flatMap(adminUsers -> {
+                    if (!adminUsers.isEmpty()) {
+                        LOGGER.debug("Revoking write rights for resource {} from admins: {}", id.value(), adminUsers);
+                        return calDavClient.revokeWriteRights(currentResource.domain(), currentResource.id(), adminUsers)
+                            .doOnError(error -> LOGGER.error("Error revoking write rights for resource {}", id.value(), error))
+                            .then(resourceDAO.softDelete(id));
+                    }
+                    LOGGER.debug("No admin users found for resource {}, proceeding with soft delete.", id.value());
+                    return resourceDAO.softDelete(id);
+                }))
+            .doOnSuccess(v -> res.status(204))
+            .thenReturn("")
+            .onErrorResume(e -> {
+                if (e instanceof ResourceNotFoundException || e instanceof IllegalArgumentException) {
```

Comment:
IllegalArgumentException is not resourceNotFound

Error management should not lie.

Webadmin handles already IllegalArgumentException